pkg/gatherers/conditional/gather_api_request_count.go

```diff
@@ -0,0 +1,106 @@
+package conditional
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// APIRequestCount defines a type used when marshaling into JSON
+type APIRequestCount struct {
+	ResourceName        string `json:"resource"`
+	RemovedInRelease    string `json:"removed_in_release"`
+	TotalRequestCount   int64  `json:"total_request_count"`
+	LastDayRequestCount int64  `json:"last_day_request_count"`
+}
+
+// BuildGatherApiRequestCounts creates a gathering closure which collects API requests counts for the
+// resources mentioned in the alert provided as a string parameter
+// Params is of type AlertIsFiringConditionParams:
+//   - alert_name string - name of the firing alert
+//
+// * Location in archive: conditional/api_request_counts.json
+// * Since versions:
+//   * 4.10+
+func (g *Gatherer) BuildGatherAPIRequestCounts(paramsInterface interface{}) (gatherers.GatheringClosure, error) {
+	params, ok := paramsInterface.(GatherAPIRequestCountsParams)
+	if !ok {
+		return gatherers.GatheringClosure{}, fmt.Errorf(
+			"unexpected type in paramsInterface, expected %T, got %T",
+			GatherAPIRequestCountsParams{}, paramsInterface,
+		)
+	}
+
+	return gatherers.GatheringClosure{
+		Run: func(ctx context.Context) ([]record.Record, []error) {
+			records, err := g.gatherAPIRequestCounts(ctx, params.AlertName)
+			if err != nil {
+				return records, err
+			}
+			return records, nil
+		},
+		CanFail: canConditionalGathererFail,
+	}, nil
+}
+
+func (g *Gatherer) gatherAPIRequestCounts(ctx context.Context, alertName string) ([]record.Record, []error) {
+	var resources []string
```

Comment:
Maybe better to make it a set to improve performance of the for loop lil bit

Reply:
you mean the https://github.com/openshift/insights-operator/pull/492/files#diff-768929cfc4894c47306f40d5abfcac538dd69ef7d32f7b0a347dc31ce6919660R79 right? Yeah I think I can use a set. Thanks